TMDbLib/Rest/RestRequest.cs

```diff
@@ -201,13 +203,21 @@ private HttpRequestMessage PrepRequest(HttpMethod method)
 
             Debug.Assert(timesToTry >= 1);
 
+            TMDbStatusMessage statusMessage = null;
+
             do
             {
                 using (HttpRequestMessage req = PrepRequest(method))
                 {
                     HttpResponseMessage resp =
                         await _client.HttpClient.SendAsync(req, cancellationToken).ConfigureAwait(false);
 
+                    if (!resp.IsSuccessStatusCode)
+                    {
+                        statusMessage =
+                            JsonConvert.DeserializeObject<TMDbStatusMessage>(await resp.Content.ReadAsStringAsync());
+                    }
+
```

Comment:
Will there _always_ be a Status message object in erronous responses?

Reply:
Aha. Just read #270 - we've seen HTML before. 

Perhaps we should _also_ check the Content-Type on the response. If it is _not_ JSON (application/json), it's an error, and we should throw the `GeneralException`.

Reply:
Hmm, not necesarily. An issue was reported were the API returned HTML instead of JSON. Not sure how to handle it exactly.

Reply:
Check for the Content-Type. I imagine that _after_ checking for 429 responses, we check if the Content Type is json. If it isn't, then it's an error no matter what the status code is.

Reply:
You can also move the parsing down to the other if..

Reply:
Want to throw a specific exception for that?

Reply:
Hmm. For response other JSON - no. It's an error, yes, but it should feed back in to the rest of the loop. 

We just should not even attempt to deserialize a status message (or an object) - if the content type is not json.